### PR TITLE
37 get services

### PR DESCRIPTION
**Pull request by fares98**

### CODE REVIEW

File: server/db/config/build.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const connection = require('./connection');
 
 const dbBuild = () => {
   const sql = readFileSync(join(__dirname, 'build.sql')).toString();
-  return connection.query(sql);
+  return connection.query(sql).then(console.log('Done')).catch((err => console.log(err)));
```
Comment (mossa-Sammer) [inline server/db/config/build.js]: catch errors when you call the function
```suggestion
  return connection.query(sql);
```

Comment (mossa-Sammer) [inline server/db/config/build.js]: catch the errors when you make the connection , not here

File: server/db/config/build.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ const connection = require('./connection');
...
 };
 
+dbBuild();
```
Comment (mossa-Sammer) [inline server/db/config/build.js]: we don' need to call dbBuild function here , we need it for the tests 
so every database test you will call the function in the test file

File: server/db/config/connection.js [hunk 3]
```diff
@@ -14,8 +14,8 @@ switch (process.env.NODE_ENV) {
   case 'development':
     dbUrl = process.env.DEV_DB_URL;
     break;
-  default:
-    throw new Error('Database Url Not Found!');
+  // default:
+  //   throw new Error('Database Url Not Found!');
```
Comment (mossa-Sammer) [inline server/db/config/connection.js]: if there is no database url , we should throw this error
```suggestion
    throw new Error('Database Url Not Found!');
```

File: server/db/queries/getServices.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const connection = require('./../config/connection');
+
+const getServices = () => {
+  connection.query(
+    'select name from services',
+    // 'select name, imgUrl(if exist) from services',
```
Comment (mossa-Sammer) [inline server/db/queries/getServices.js]: remove commented code

File: server/controllers/index.js [hunk 5]
```diff
@@ -4,9 +4,11 @@ const logout = require('../controllers/routes/getLogout');
 
 const router = express.Router();
 
+const { routeServices } = require('./routes/getServices');
```
Comment (hashemt) [inline server/controllers/index.js]: delete this line and add the function in the next line

File: server/controllers/index.js [hunk 6]
```diff
@@ -4,9 +4,11 @@ const logout = require('../controllers/routes/getLogout');
...
 const { getUserJobs } = require('./routes');
```
Comment (hashemt) [inline server/controllers/index.js]:
```suggestion
const { getUserJobs, routeServices } = require('./routes');
```

File: server/controllers/routes/getServices.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const { getServices } = require('../../db/queries/getServices');
+
+exports.routeServices = async (req, res) => {
+  try {
+    const services = await getServices();
+    res.send({ data: services.rows });
+  } catch (err) {
+    console.log(err);
```
Comment (hashemt) [inline server/controllers/routes/getServices.js]:
```suggestion
    next(err);
```

File: server/controllers/routes/getServices.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+const { getServices } = require('../../db/queries/getServices');
+
+exports.routeServices = async (req, res) => {
```
Comment (hashemt) [inline server/controllers/routes/getServices.js]:
```suggestion
exports.routeServices = async (req, res, next) => {
```

File: server/db/config/build.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ const connection = require('./connection');
 
 const dbBuild = () => {
   const sql = readFileSync(join(__dirname, 'build.sql')).toString();
-  return connection.query(sql);
+  return connection.query(sql).then(console.log('Done')).catch(((err) => console.log(err)));
```
Comment (hashemt) [inline server/db/config/build.js]:
```suggestion
  return connection.query(sql);
```

File: server/db/config/connection.js [hunk 10]
```diff
@@ -18,6 +18,7 @@ switch (process.env.NODE_ENV) {
     throw new Error('Database Url Not Found!');
 }
 
+
```
Comment (hashemt) [inline server/db/config/connection.js]: line !!!

File: server/db/config/insertFakeData.js [hunk 11]
```diff
@@ -8,4 +8,6 @@ const insertFakeData = () => {
   return connection.query(sql);
 };
 
+insertFakeData();
```
Comment (hashemt) [inline server/db/config/insertFakeData.js]: delete this line

File: server/db/queries/getServicesById.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+const connection = require('./../config/connection');
+
+
```
Comment (hashemt) [inline server/db/queries/getServicesById.js]: line !!!

File: server/tests/queries/index.js [hunk 13]
```diff
@@ -1 +1,8 @@
+const tape = require('tape');
+const { getServices } = require('./getServices');
 require('./getJob');
+
+tape('Simple pass test', (t) => {
+  t.equal(1, 1, 'one should equal one');
+  t.end();
+});
```
Comment (hashemt) [inline server/tests/queries/index.js]: delete simple pass test

File: server/tests/routes/getServices.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+const test = require('tape');
+const supertest = require('supertest');
+
+const { routeServices } = require('../../controllers/routes/getServices');
+const app = require('./../../app');
```
Comment (hashemt) [inline server/tests/routes/getServices.js]:
```suggestion
const app = require('../../app');
```

File: server/tests/routes/index.js [hunk 15]
```diff
@@ -1,2 +1,7 @@
 require('./getJob');
+require('./getServices');
 
+tape('Simple pass test', (t) => {
+  t.equal(1, 1, 'one should equal one');
+  t.end();
+});
```
Comment (hashemt) [inline server/tests/routes/index.js]: delete simple test

**Pull request closed without merging**